lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced/search-date-range-advanced.component.ts

```diff
@@ -0,0 +1,163 @@
+    ngOnInit(): void {
+        this.dateFormatConfig.display.dateInput = this.dateFormat;
+        this.convertedMaxDate = endOfDay(this.maxDate && this.maxDate !== 'today' ?
+            parse(this.maxDate, this.dateFormat, new Date()) : new Date());
+        this.userPreferencesService
+            .select(UserPreferenceValues.Locale)
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(locale => this.dateAdapter.setLocale(DateFnsUtils.getLocaleFromString(locale)));
+        this.form.controls.dateRangeType.valueChanges.pipe(takeUntil(this.destroy$))
+            .subscribe((dateRangeType) => this.updateValidators(dateRangeType));
+        this.form.valueChanges.pipe(takeUntil(this.destroy$))
+            .subscribe(() => this.onChange());
+    }
+
+    ngOnDestroy() {
+        this.destroy$.next();
+        this.destroy$.complete();
+    }
+
+    private updateValidators(dateRangeType: DateRangeType) {
+        switch(dateRangeType) {
+            case DateRangeType.BETWEEN:
+                this.betweenStartDateFormControl.setValidators(Validators.required);
+                this.betweenEndDateFormControl.setValidators([Validators.required, this.endDateValidator]);
+                this.form.controls.inLastValue.clearValidators();
+                break;
+            case DateRangeType.IN_LAST:
+                this.form.controls.inLastValue.setValidators(Validators.required);
+                this.betweenStartDateFormControl.clearValidators();
+                this.betweenEndDateFormControl.clearValidators();
+                break;
+            case DateRangeType.ANY:
+                this.form.controls.inLastValue.clearValidators();
+                this.betweenStartDateFormControl.clearValidators();
+                this.betweenEndDateFormControl.clearValidators();
+                break;
+        }
+        this.betweenStartDateFormControl.updateValueAndValidity();
+        this.betweenEndDateFormControl.updateValueAndValidity();
+        this.form.controls.inLastValue.updateValueAndValidity();
+    }
+
+    private onChange(): void {
+        if (this.form.valid) {
+            this.changed.emit(this.form.value);
+        }
+        this.valid.emit(this.form.valid);
+    }
+
+    dateChanged(event: Event, formControl: UntypedFormControl) {
+        if (event?.target['value']?.trim()) {
+            const date = parse(event.target['value'], this.dateFormat, new Date());
+            if(!isValid(date)) {
+                formControl.setErrors({
+                    ...formControl.errors,
+                    required: false,
+                    invalidDate: true
+                });
+            } else {
+                formControl.setErrors({
+                    ...formControl.errors,
+                    invalidDate: false
+                });
+                formControl.setValue(date);
+            }
+        }
+    }
+
+    onLastDateValueChanged(event: Event) {
+        let value: string = event.target['value'];
```

Comment:
should be const

Reply:
These has been resolved in [this](https://github.com/Alfresco/alfresco-ng2-components/pull/8714/commits/bdaa0e52fd3a31ba7c3802c6cdc6a27a733d6057) commit